src/index.js

```diff
@@ -0,0 +1,28 @@
+const fs = require('fs');
+const tempy = require('tempy');
+const request = require('request');
+const retriableErrorCodes = ['ECONNRESET', 'ETIMEOUT', 'ESOCKETTIMEDOUT'];
+
+module.exports = function retryDownload (retries, ...args) {
+  return new Promise((resolve, reject) => {
+    const filename = tempy.file();
+    const writable = fs.createWriteStream(filename);
+
+    writable.once('finish', () => {
+      return resolve(filename);
+    });
+
+    const r = request(...args);
+
+    r.once('response', function (response) {
+      if (!/^2[0-9][0-9]$/.exec(response.statusCode)) {
```

Comment:
should we maybe be retrying on 500 errors as well? 503, 504?

Reply:
maybe add a `retriableResponseCodes` array? maybe make both `retriable*` arrays configurable with our own list of default options?